### ISSUE TITLE
Possible fix for the .meta tiddler handling problem

### ISSUE THREAD

**Issue by zimiarh** [OokTech/TW5-MultiUser#13]
I have been working on making external image to work in the last couple days. The problem I tried to solve is this: when saving and uploading changes of a tiddler that is recognized to have metadata, the content of the tiddler in the clients will not get updated properly. I am not sure if this is the problem mentioned in the note of the README.

After bumping my head against the wall for a few times, I finally found the cause and a workaround to it. The cause is something to do with how the tiddlers on client side got updated. When a tiddler got changed in a client, the change is submitted to the server and then broadcasted to all the clients. However, the tiddler got broadcasted is not the one submitted in the first place. When the server receives a tiddler from a client, it saves the tiddler to its filesystem. After that, a filesystem monitor got triggered and reads (supposedly) the latest version of the tiddler back into wiki and broadcasts this tiddler to all the clients.

When a tiddler submitted to the server does not have any metadata, this pipeline runs smoothly. However, if the tiddler has metadata, this pipeline does not work out as expected because its metadata are not read back into wiki correctly when the filesystem monitor tries to broadcast the changes of the tiddler. 

Here is how the bug happens in detail. When a tiddler with metadata is submitted to the server, its main body (without metadata) got saved to filesystem first (before its metadata), and such an action triggers the filesystem monitor to broadcast the changes. So the filesystem monitor loads the tiddler from file, in addition with its metadata. However, at this point, the metadata of the tiddler is most likely not being saved to the filesystem yet, because the metadata are saved in the callback of the action of saving the tiddler. Thus, the tiddler loaded from the filesystem is actually outdated and so the one got broadcasted to and shown in the clients.

The proposed workaround is pretty simple: just reverse the order of saving the tiddler data. Saving metadata first then the tiddler itself seems to solve the problem. However, I am not quite sure if this is a proper and solid way to solve it as I am not familiar with node.js and asynchronous stuffs. 

I do have a commit with this on my own fork, but I am afraid that it cannot get merged because my fork is way too different from the origin.

## TL; DR
In the `WebsocketAdaptor.saveTiddler`, saving metadata first then the tiddler itself seems to solve the problem mentioned in the first note of the README

**Comment by inmysocks** [comment]
The multi-user plugin can server multiple wikis now and that made the fix a bit more involved than your solution, but you gave me the start that I needed to work it out so thank you!
